documentation/fields.md

```diff
@@ -0,0 +1,162 @@
+# Fields
+
+Fields is an object of key/value pairs - each pair the name of a field, and a configuration options object.
+```js
+module.exports = { /... key/value pairs go here .../ };
+```
+The configuration options can include properties such as `validate`, `label`, `className`, and others, as documented in each of the examples below.
+
+Note: `validate` pertains to server-side validation.
+
+Its important to remember that the fields themselves do not render as HTML, but make available the properties that might be necessary to create an HTML field. The HOF stack uses [hmpo-template-mixins](https://github.com/UKHomeOffice/passports-template-mixins) - exported as `mixins` - along with the field names and configuration options to render partial HTML templates.
+
+For more information about `fields` see [Form Wizard](https://github.com/UKHomeOffice/passports-form-wizard), and [Form Controller](https://github.com/UKHomeOffice/passports-form-controller#validators).
+
+## Usage
+
+The following are examples of various fields and their configuration options
+
+### Radio buttons
+
+```js
+'my-radio-buttons': {
+  options: [{
+    value: 'yes',
+    label: 'Yes'
+  }, {
+    value: 'no',
+    label: 'No',
+    toggle: 'myToggleField'
+  }],
+  validate: ['required'],
+  className: ['inline']
+}
+```
+
+- `my-radio-buttons` has two options.
+- `options` is a collection of items, each corresponding to an option in a radio control, and each with its own `label` and `value` property.
+- option 'yes' has a name of 'my-radio-buttons' and an id of 'my-radio-buttons-yes'.
+- option 'no' has a name of 'my-radio-buttons' and an id of 'my-radio-buttons-no'.
+- `toggle` points to an id and can be used show the html field with "id='myToggleField'" when the radio option "no" is selected'
+- `validate` takes an array of one or more, optional validators, [listed here](https://github.com/UKHomeOffice/passports-form-controller/blob/master/lib/validation/validators.js).
+- `className` takes an array of html class names that can be mapped to the corresponding HTML element.
+
+
+### Date fields
+
+```js
+'my-date': {
+  legend: 'Date',
+  hint: 'For example, 31 3 1970'
+},
+'my-date-day': {
+  validate: ['required', 'numeric'],
```

Comment:
Having a play with this having 'required' on each of the date field produces slightly funny behaviour. It works fine if we put the 'required' validation on the 'my-date' field though (it is entirely possible I've messed something up to get this behaviour...).

The behaviour is:
- If no date fields are filled in it produces an error but doesn't give any error message
- If some but not all of the date fields are filled in it works fine and produces the correct error message
- If all date fields are filled in then it works fine

Suggestion is we change the example to have the 'required' validation on the 'my-date' field